coffeecard/CoffeeCard.Library/Services/AccountService.cs

```diff
@@ -256,6 +248,60 @@ public async Task<bool> RecoverUserAsync(string token, string newPassword)
             await _context.SaveChangesAsync();
             return true;
         }
+        
+        public bool UserExists(string email)
+        {
+            return _context.Users.Any(x => x.Email == email);
+        }
+
+        public void AnonymizeAccount(string email)
+        {
+            var user = _context.Users.First(x => x.Email == email);
+            
+            var claims = new[]
+            {
+                new Claim(ClaimTypes.Email, user.Email), new Claim(ClaimTypes.Name, user.Name),
+                new Claim(ClaimTypes.Role, "verification_token")
+            };
+            var verificationToken = _tokenService.GenerateToken(claims);
+            
+            _emailService.SendVerificationEmailForDeleteAccount(user, verificationToken);
+        }
+        
+        public bool VerifyAnonymization(string token)
+        {
+            Log.Information($"Trying to verify deletion with token: {token}");
+
+            var user = VerifyTokenClaimAndUser(token);
+
+            AnonymizeUser(user);
+            return _context.SaveChanges() > 0;
```

Comment:
Will this not always return false since we already saved the changes in `AnonymizeUser`? Also, I think returning `void` is just fine since an exception should be thrown if we can not save the user etc.